hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -150,40 +201,139 @@ def _load_dataset(self, tiles_dataset: TilesDataset, stage: str, shuffle: bool)
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for random) subset
+            Defaults to ``min`` (which assumes background is high value)
+    """
 
-    def test_dataloader(self) -> DataLoader:
-        return self._get_dataloader(self.test_dataset, 'test', shuffle=True, **self.dataloader_kwargs)
+    def __init__(
+        self,
+        level: Optional[int] = 0,
+        tile_count: Optional[int] = None,
+        tile_size: Optional[int] = 224,
+        step: Optional[int] = None,
+        random_offset: Optional[bool] = False,
+        pad_full: Optional[bool] = False,
+        background_val: Optional[int] = 255,
+        filter_mode: Optional[str] = "min",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.level = level
+        self.tile_count = tile_count
+        self.tile_size = tile_size
+        self.step = step
+        self.random_offset = random_offset
+        self.pad_full = pad_full
+        self.background_val = background_val
+        self.filter_mode = filter_mode
+
+    def prepare_data(self) -> None:
+        if self.precache_location != CacheLocation.NONE:
+            self._load_dataset(self.train_dataset, stage="train")
+            self._load_dataset(self.val_dataset, stage="val")
+            self._load_dataset(self.test_dataset, stage="test")
+
+    def _load_dataset(self, slides_dataset: SlidesDataset, stage: str) -> Dataset:
+        dataset_pickle_path = self._dataset_pickle_path(stage)
+
+        if dataset_pickle_path and dataset_pickle_path.is_file():
+            if self.precache_location == CacheLocation.CPU:
+                memory_location = torch.device("cpu")
+                print(f"Loading dataset from {dataset_pickle_path} into {memory_location}")
+            else:
+                # by default torch.load will reload on the same device it was saved from
+                memory_location = None  # type: ignore
+
+            with dataset_pickle_path.open("rb") as f:
+                return torch.load(f, map_location=memory_location)
+
+        base_transform = Compose(
+            [
+                LoadImaged(
+                    keys=slides_dataset.IMAGE_COLUMN,
+                    reader=WSIReader,
+                    backend="cucim",
+                    dtype=np.uint8,
+                    level=self.level,
+                    image_only=True,
+                ),
+                TileOnGridd(
+                    keys=slides_dataset.IMAGE_COLUMN,
+                    tile_count=self.tile_count,
+                    tile_size=self.tile_size,
+                    step=self.step,
+                    random_offset=self.random_offset,
+                    pad_full=self.pad_full,
+                    background_val=self.background_val,
+                    filter_mode=self.filter_mode,
+                    return_list_of_dicts=True,
+                ),
+            ]
+        )
```

Comment:
Just noting that I think we will need some design changes to integrate with deepmil (but we can tackle in the next PR):

- we will need to put together in a tensor the tiles and the labels like here https://github.com/Project-MONAI/tutorials/blob/622d460e0845cebff5f32022b2ef35448b61a1ee/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py#L369 
- we will need the ability to pass different transformations for training and testing purposes